CMakeLists.txt

```diff
@@ -1,32 +1,47 @@
 cmake_minimum_required(VERSION 3.16)
-project(Noxis)
+project(Noxis C)
 
-set(CMAKE_C_FLAGS "-Wall -Wpedantic -Wextra -Werror")
-set(CMAKE_C_STANDARD 11)
+include(cmake/base.cmake)
+include(cmake/warnings.cmake)
 
-set(CMAKE_BINARY_DIR ${PROJECT_SOURCE_DIR})
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
-set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+find_package(SDL2 REQUIRED)
 
-#set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
+add_executable(${PROJECT_NAME})
 
-find_package(SDL2 REQUIRED)
+target_sources(
+    ${PROJECT_NAME}
+    PRIVATE
+        include/utils.h
+        src/utils.c
 
-set(LINK_LIBS ${SDL2_LIBRARIES})
```

Comment:
There is no need for a global variable. Also, the imported `SDL2::SDL2` target should be used.